### ISSUE TITLE
 Scaling questions in multi-exit loss configurations.

### ISSUE THREAD

**Issue by seominseok0429** [mary-phuong/multiexit-distillation#3]
hello. l am a college student studying deep learning in Korea.

i read your paper impressed.

i was wondering while reading the paper. did you scale each loss (depending on network depth) when configuring a multi-exit loss?

for example, ( depth = exit1 < exit2 < exit3) 

exit1 , exit2, exit3 = model(input)

loss1  = criterion(exit1, targets)
loss2 = criterion(exit2, targets) 
loss3 = criterion(exit3, targets)

no scale

total_loss = loss1 + loss2 + loss3

scale

total_loss = 0.1*loss1 + 0.2*loss2 + 0.7*loss3

If you didn't scale, can you tell me why?

i wanted to solve it by myself, but i can't solve it.
i'm really sorry.

best regards

**Comment by mary-phuong** [comment]
Hi.
No, I didn't scale the losses, but I think in principle you should scale
the losses based on the relative importance of the exits (more important
exits should get higher weights).
The reason I don't scale is that in my experiments, there is no such prior
importance information.

**Comment by seominseok0429** [comment]
Thank you very much for your kind reply.
